android-components/components/browser/engine-gecko/src/test/java/mozilla/components/test/ReflectionUtils.kt

```diff
@@ -4,18 +4,38 @@
 
 package mozilla.components.test
 
+import java.lang.invoke.MethodHandles
 import java.lang.reflect.Field
 import java.lang.reflect.Modifier
+import java.security.AccessController
+import java.security.PrivilegedAction
 
 object ReflectionUtils {
     fun <T : Any> setField(instance: T, fieldName: String, value: Any?) {
-        val originField = instance.javaClass.getField(fieldName)
+        val mapField = AccessController.doPrivileged(
+            PrivilegedAction {
```

Comment:
The comment on this class says "Legacy security code; do not use", and "Android-changed: Stubbed the implementation.  Android doesn't support SecurityManager."

The android comment we can ignore since this is a test utils class that will not be run on Android, but I'm not sure about the first one.

The [javadocs also say](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/PrivilegedAction.html), we ought to use the `PrivilegedExceptionAction` instead for calling something that throws.

Reply:
I switch to using `PrivilegedExceptionAction`